test_runner/regress/test_pageserver_generations.py

```diff
@@ -210,9 +211,30 @@ def test_generations_upgrade(neon_env_builder: NeonEnvBuilder):
 
     env.pageserver.start(overrides=('--pageserver-config-override=control_plane_api=""',))
 
-    env.neon_cli.create_tenant(
-        tenant_id=env.initial_tenant, conf=TENANT_CONF, timeline_id=env.initial_timeline
-    )
+    try:
+        env.neon_cli.create_tenant(tenant_id=env.initial_tenant, conf=TENANT_CONF)
+    except Exception:
+        # The initial tenant creation will likely fail because the external node registration
```

Comment:
Can we add a `env.storage_controller.node_configure` call to set the node available on line 213, instead of taking this exceptional path?

Reply:
We can and it should work, although this is a scenario that we don't test somewhere else.

Reply:
Does this still apply with your most recent changes?  (now that we mark things active in re-attach)

If we do still want to test this scenario, let's do it somewhere in a test that's related to the sharding service -- it's a bit strange to do it in this generations test.

Reply:
> Does this still apply with your most recent changes?

Yes. We start the pageserver without the control plane api (`--pageserver-config-override=control_plane_api=""`), so it doesn't call re-attach which means it doesn't hit the new code path.

Reply:
Took on your original suggsetion. My version of the test doesn't work without background rescheduling anyway.